docs/lang_spec/Cadence Language Specification.md

```diff
@@ -0,0 +1,5798 @@
+
+By organizing and defining the fundamental building blocks of Cadence programs, the Declarations section provides the framework necessary to ensure that code is robust, secure, and well-structured.
+
+- **EBNF**:
+  ```ebnf
+  declaration
+      structDeclaration
+    | resourceDeclaration
+    | contractDeclaration
+    | structInterfaceDeclaration
+    | resourceInterfaceDeclaration
+    | contractInterfaceDeclaration
+    | functionDeclaration
+    | variableDeclaration
+    | eventDeclaration
+    | transactionDeclaration
+    | pragmaDeclaration
+    | entitlementDeclaration
+    | entitlementMappingDeclaration
+    ;
+
+  ```
+
+#### Struct Declarations
+
+In Cadence, a struct is a composite type that allows the grouping of related data fields into a single entity. Structs are value types, meaning that they are copied when passed or assigned. This behavior makes them ideal for situations where independent copies of a data structure with separate state are required.
+
+Structs in Cadence are declared using the struct keyword, followed by an identifier (the name of the struct), optional conformances to interfaces, and a block containing the struct’s fields and any associated functions. The access level of a struct and its members is controlled by access modifiers, allowing developers to determine the visibility and accessibility of struct elements.
+
+- **Syntax**:
+  - The basic syntax for a struct declaration is as follows:
+    ```cadence
+    access(struct) struct StructName {
+      access(struct) let constantField: Type
+      access(struct) var variableField: Type
+      
+      init(parameterName: Type) {
+        // Initialization of fields
+        self.constantField = parameterValue
+        self.variableField = anotherValue
+      }
+      
+      fun someFunction() {
+        // Function body
+      }
+    }
+
+    ```
+    - **access**: Specifies the access control for the struct and its members (e.g., pub, access(contract), access(self)).
+    - **StructName**: The name of the struct.
+    - **Fields**: Declared with the let or var keywords. Constant fields (let) are immutable after initialization, while variable fields (var) can be modified after the struct is created.
+    - **Initializer**: The init function is responsible for initializing all fields in the struct. It ensures that each field is assigned a value before the struct is used.
+    - **Functions**: Structs can contain functions, which operate on their fields and are defined within the struct body.
+
+- **Example**:
+  ```cadence
+  pub struct User {
+    pub let id: Int
+    pub var name: String
+    
+    init(id: Int, name: String) {
+      self.id = id
+      self.name = name
+    }
+     
+    pub fun updateName(newName: String) {
+      self.name = newName
+    }
+  }
+  ```
```

Comment:
Note that in Cadence 1.0 the access specifiers `pub`, `pub(set)` and `priv` got removed. See "pub and priv Access Modifiers Got Removed" in https://github.com/onflow/cadence/releases/tag/v1.0.0

Reply:
I plan to remove many of the examples. They may be better fitted to another document. I will definitely remove the ones that use `pub` and `priv`.